src/Http/Controllers/ExcelController.php

```diff
@@ -20,14 +20,78 @@ class ExcelController extends Controller
      */
     public function download(Request $request, ResponseFactory $response): BinaryFileResponse
     {
+        if(!$this->canAccess($request)) {
+            abort(404);
+        }
+
         $data = $this->validate($request, [
             'path'     => 'required',
             'filename' => 'required',
         ]);
 
-        return $response->download(
-            $data['path'],
-            $data['filename']
-        )->deleteFileAfterSend($shouldDelete = true);
+        try{
+            $download = $response->download(
+                $data['path'],
+                $data['filename']
+            )->deleteFileAfterSend($shouldDelete = true);
+        }catch (\Exception $exception) {
+            abort(403);
+        }
+        return $download;
+    }
+
+    /**
+     * Check if file can be acessed
+     *
+     * @param Request $request
+     * @return boolean
+     */
+    protected function canAccess(Request $request):bool
+    {
+        $canAccess = true;
+        $pathInfo = pathinfo($request->input('filename'));
+
+        if(!$this->validadePath($request->input('path'))) {
+            $canAccess = false;
+        }
+        if($canAccess) {
+            if(!$this->checkFileExtension($pathInfo['extension'])) {
+                $canAccess = false;
+            }
+        }
+        return $canAccess;
+    }
+
+    /**
+     * Check if file extension is valid
+     *
+     * @param string $fileExtension
+     * @return boolean
+     */
+    protected function checkFileExtension(string $fileExtension):bool
+    {
+        $isValid = true;
+        $validFileExtensions = ['csv', 'xlsx', 'xls'];
+        if(!in_array($fileExtension, $validFileExtensions)) {
+            $isValid = false;
+        }
+        return $isValid;
+    }
+
+    /**
+     * Check if path is valid
+     *
+     * @param string $path
+     * @return boolean
+     */
+    protected function validadePath(string $path):bool
+    {
+        $isValid = false;
+        $strRealPath = realpath($path);
+
+        if (strpos($strRealPath, base_path().'/storage') === 0) {
```

Comment:
By default, `$path` would use /`tmp/`, not Laravel storage. Is there any other modification that you forgot to commit?